src/lib/components/Dropdown.svelte

```diff
@@ -0,0 +1,32 @@
+<script lang="ts">
+  import { slide } from 'svelte/transition'
+  import Button from '$lib/elements/Button.svelte'
+  import caret from '$lib/icons/caret'
+
+  export let label = ''
+  let dropdownOpen = false
+</script>
+
+<!-- svelte-ignore a11y-click-events-have-key-events -->
+<div class="relative inline-block text-left">
+  <Button
+    small={true}
+    text={label}
+    on:click={() => {
+      dropdownOpen = !dropdownOpen
+    }}
+  >
+    <div slot="iconRight" class="w-5 ml-2 mb-[2px]" class:-rotate-180={!dropdownOpen}>
+      {@html caret}
+    </div>
+  </Button>
+  {#if dropdownOpen}
+    <div
+      in:slide|local={{ duration: 250 }}
+      out:slide|local={{ duration: 250 }}
+      class="absolute left-0 z-10 mt-2 bg-white dark:bg-neutral-900 dark:text-white border rounded-md shadow-lg"
+    >
+      <slot />
+    </div>
+  {/if}
+</div>
```

Comment:
```suggestion
<!-- svelte-ignore a11y-click-events-have-key-events -->
<div on:click|stopPropagation class="relative inline-block text-left">
  <Button
    small={true}
    text={label}
    on:click={() => {
      dropdownOpen = !dropdownOpen
    }}
  >
    <div slot="iconRight" class="w-4 ml-2 transition-all" class:-rotate-180={dropdownOpen}>
      {@html caret}
    </div>
  </Button>
  {#if dropdownOpen}
    <div
      use:clickOutside={() => (dropdownOpen = false)}
      transition:slide|local={{ duration: 250 }}
      class="absolute left-0 z-10 mt-2 bg-white dark:bg-neutral-900 dark:text-white border rounded-md shadow-lg"
    >
      <slot />
    </div>
  {/if}
</div>
```

A couple of suggestions here:

- The caret was flipped the wrong way
- Change the styles of the caret to fit other UI elements throughout the app
- Can use `transition` directive instead of defining the same transitions for both `in` and `out`
- We can use the `clickOutside` directive so that when you click outside of the dropdown, it automatically closes.